src/include/distributed/version_compat.h

```diff
@@ -38,6 +38,9 @@
 #define PreventInTransactionBlock PreventTransactionChain
 #define DatumGetJsonbP(d) DatumGetJsonb(d)
 
+/* following is introduced in PG11 */
+#define ConstraintRelidTypidNameIndexId ConstraintRelidIndexId
```

Comment:
OK, so I know this sounds backwards of what we _usually_ want to do, but can you define `ConstraintRelidIndexId` for PostgreSQL 11 rather than defining `ConstraintRelidTypidNameIndexId` for 9.6 and 10?

Yes, we'd remove this `#define` eventually, but the important thing here is that `ConstraintRelidTypidNameIndexId` can be used to look up an index by `relid` (as it's a prefix of the index), but `ConstraintRelidIndexId` is _not_ actually a valid stand-in for `ConstraintRelidTypidNameIndexId` (because it's an index of only a single column, not all three).

Because of this, I think the `#define` you have here is actually misleading (it pretends to backfill some behavior that's actually not implementable, namely this three-column index). Instead, we can make an alias for the name in PostgreSQL 11 to reflect the fact that this new index is still usable in the old manner (looking up by relid).